context/app/static/js/components/workspaces/WorkspaceField/WorkspaceField.tsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { FieldValues, Path, PathValue, useController, UseControllerProps } from 'react-hook-form';
+import Box from '@mui/material/Box';
+import TextField, { TextFieldProps } from '@mui/material/TextField';
+
+interface WorkspaceFieldProps<FormType extends FieldValues>
```

Comment:
After seeing how you have to assert the form type if you don't pass the `value`, I'm not as much of a fan of this as I was initially; I can revisit this to see if we can make usage less of a pain.

I wonder if we could substitute passing some of these through props with https://react-hook-form.com/docs/useformcontext and https://react-hook-form.com/docs/formprovider?